bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -0,0 +1,204 @@
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private Set<ScriptEngineProvider> scriptEngineProviders = new HashSet<>();
+    private HashMap<String, ScriptEngineContainer> loadedScriptEngineInstances = new HashMap<>();
+    private HashMap<String, ScriptEngineProvider> supportedLanguages = new HashMap<>();
+    private GenericScriptEngineProvider genericProvider = new GenericScriptEngineProvider();
+    private ScriptExtensionManager scriptExtensionManager;
+
+    public ScriptEngineManagerImpl() {
+        logger.info("ScriptManager loading...");
+    }
+
+    public void activate(BundleContext bundleContext) {
+    }
+
+    public void addScriptEngineProvider(ScriptEngineProvider provider) {
+        this.scriptEngineProviders.add(provider);
+
+        for (String language : provider.getLanguages()) {
+            this.supportedLanguages.put(language, provider);
+        }
+    }
+
+    public void removeScriptEngineProvider(ScriptEngineProvider provider) {
+        this.scriptEngineProviders.remove(provider);
+    }
+
+    public void setScriptExtensionManager(ScriptExtensionManager scriptExtensionManager) {
+        this.scriptExtensionManager = scriptExtensionManager;
+    }
+
+    public void unsetScriptExtensionManager(ScriptExtensionManager scriptExtensionManager) {
+        this.scriptExtensionManager = null;
+    }
+
+    @Override
+    public boolean isSupported(String fileExtension) {
+        return findProvider(fileExtension) != null;
+    }
+
+    /*
+     * @Override
+     * public ScriptEngine loadScript(File file) {
+     * try (InputStreamReader streamReader = new InputStreamReader(new FileInputStream(file), UTF_8)) {
+     * return loadScript(uniqueId(file), streamReader);
+     * } catch (IOException ex) {
+     * logger.error("", ex);
+     * }
+     *
+     * return null;
+     * }
+     */
+
+    @Override
+    public ScriptEngineContainer createScriptEngine(String fileExtension, String scriptIdentifier) {
+        ScriptEngineContainer result = null;
+        ScriptEngineProvider engineProvider = findProvider(fileExtension);
+
+        if (engineProvider == null) {
+            logger.error("loadScript(): scriptengine for language '{}' could not be found for identifier: {}",
+                    fileExtension, scriptIdentifier);
+        } else {
+            try {
+                ScriptEngine engine = engineProvider.createScriptEngine(fileExtension);
+                HashMap<String, Object> scriptExManager = new HashMap<>();
+                result = new ScriptEngineContainer(engine, engineProvider, scriptIdentifier);
+                scriptExManager.put("ScriptExtension",
+                        new ScriptExtensionManagerWrapper(result, scriptExtensionManager));
+                engineProvider.scopeValues(engine, scriptExManager);
+                scriptExtensionManager.importDefaultPresets(engineProvider, engine, scriptIdentifier);
```

Comment:
I'm seeing NPEs on this line after the latest commits. I haven't been able to determine why scriptExtensionProvider is null.